### PR TITLE
Add support to PATCH requests in network client

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

Adding support for PATCH operations in our network client.

## How was this tested?

Automated tests.

### CODE REVIEW

File: assets/js/lib/network/network.test.js [hunk 1]
```diff
@@ -188,4 +188,23 @@ describe('networkClient', () => {
       }
     });
   });
+
+  describe('REST operations', () => {
```
Comment (jamie-suse) [inline assets/js/lib/network/network.test.js:192]: Is this test really testing anything?

Comment (nelsonkopliku) [reply assets/js/lib/network/network.test.js:192]: Nothing SUMA specific. It's pretty naive I know 😂 I can live also without it.

Comment (jamie-suse) [reply assets/js/lib/network/network.test.js:192]: I guess `axiosMock.onPatch()` is doing the heavy-lifting here and validates that the request is a PATCH request. :muscle:

I'm fine with it either way also, but I guess we should also then test the other verbs in `networkClient`, no?

Comment (nelsonkopliku) [reply assets/js/lib/network/network.test.js]: The test you were looking for is not here :smile:

**Review by dottorblaster — APPROVED** [review]
LGTM

**Pull request merged**